src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -741,25 +818,52 @@ public void withLock(boolean waitForRecovery, ChannelConsumer consumer) throws I
             }
         }
 
-        public void retryingPublishWithLock(ChannelConsumer consumer, ConnectionManagerConfiguration configuration) throws InterruptedException {
+        public void retryingPublishWithLock(ConnectionManagerConfiguration configuration, byte[] payload, ChannelPublisher publisher) throws IOException, InterruptedException {
             lock.lock();
             try {
                 Iterator<RetryingDelay> iterator = configuration.createRetryingDelaySequence().iterator();
                 Channel tempChannel = getChannel(true);
+                // add current message to the end to unify logic for sending current and redelivered messages
+                redeliveryQueue.addLast(new PublicationHolder(publisher, payload));
                 while (true) {
-                    try {
-                        consumer.consume(tempChannel);
+                    if (publishConfirmationListener.isNoConfirmationWillBeReceived()) {
+                        LOGGER.warn("Connection was closed on channel. No delivery confirmation will be received. Drain message to redelivery");
+                        publishConfirmationListener.transferUnconfirmedTo(redeliveryQueue);
+                    }
+                    PublicationHolder currentPayload = redeliveryQueue.pollFirst();
+                    if (!redeliveryQueue.isEmpty()) {
+                        LOGGER.warn("Redelivery queue size: {}", redeliveryQueue.size());
+                    }
+                    if (currentPayload == null) {
                         break;
+                    }
+                    long msgSeq = tempChannel.getNextPublishSeqNo();
+                    try {
+                        publishConfirmationListener.put(msgSeq, currentPayload);
+                        if (publishConfirmationListener.isNoConfirmationWillBeReceived()) {
+                            // will drain message to queue on next iteration
+                            continue;
+                        }
+                        currentPayload.publish(channel);
```

Comment:
Yes, we should. In reality, it didn't change anything because when `createChannel` is executed the `channel` will have the same value as `tempChannel` but you are right - we should use `tempChannel` here